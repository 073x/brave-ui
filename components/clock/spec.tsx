/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Clock from '../clock'

describe('boxedContent tests', () => {
  const baseComponent = (props?: object) => <Clock {...props} />

  it('matches the snapshot', () => {
    const component: any = shallow(baseComponent())
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component', () => {
    const wrapper = shallow(baseComponent({id: 'someId'}))
    const assertion = wrapper.find('#someId').length
    expect(assertion).toBe(1)
  })

  it('can have an id', () => {
    const wrapper = shallow(baseComponent({id: 'whatClock'}))
    const assertion = wrapper.props().id
    expect(assertion).toBe('whatClock')
  })
})
