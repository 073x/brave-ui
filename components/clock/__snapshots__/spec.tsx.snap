// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boxedContent tests matches the snapshot 1`] = `
<div
  className="sc-bdVaJa bBabaa"
  id={undefined}
>
  <span
    className="sc-bwzfXH jkxQTc"
  >
    02
    <span
      className="sc-bxivhb dPajcN"
    >
      :
    </span>
    07
  </span>
  <span
    className="sc-htpNat fpUzll"
  >
    
  </span>
</div>
`;
